core/test_network/src/lib.rs

```diff
@@ -553,43 +533,202 @@ where
             broker: Broker::new(),
         })
     }
+}
 
-    /// Starts peer with default configuration.  **IMPORTANT**: Retain
-    /// all three parameters for the scope of the test. Do not ignore
-    /// the first two elements of the tuple.
-    /// Returns its info and client for connecting to it.
-    pub fn start_test_with_runtime() -> (Runtime, Self, Client) {
-        let rt = Runtime::test();
-        let (peer, client) = rt.block_on(Self::start_test_with_permissions(
-            AllowAll.into(),
-            AllowAll.into(),
-        ));
-        (rt, peer, client)
+struct PeerArgs<W, G>
+where
+    W: WorldTrait,
+    G: GenesisNetworkTrait,
+{
+    genesis: WithGenesis<G>,
+    configuration: Option<Configuration>,
+    instruction_validator: Option<IsInstructionAllowedBoxed<W>>,
+    query_validator: Option<IsQueryAllowedBoxed<W>>,
+    temp_dir: Option<Arc<TempDir>>,
+}
+
+impl<W, G> Default for PeerArgs<W, G>
+where
+    W: WorldTrait,
+    G: GenesisNetworkTrait,
+{
+    fn default() -> Self {
+        Self {
+            genesis: WithGenesis::<G>::default(),
+            configuration: None,
+            instruction_validator: None,
+            query_validator: None,
+            temp_dir: None,
+        }
+    }
+}
+
+/// `WithGenesis` structure.
+///
+/// Options for setting up the genesis network for `PeerBuilder`.
+pub enum WithGenesis<G> {
+    /// The option means that it will create the default genesis network.
+    Default,
+    /// The option means pass nothing.
```

Comment:
not clear to me what the option means

```suggestion
    /// Pass nothing.
```

Reply:
This literally means that there is no genesis. So the Iroha peer will start without a given genesis.
I made this custom enum from `Option<Option<G>>`.